File: Army/units/rogue.cpp
#include "rogue.h"

using namespace std;

Rogue::Rogue(const string& name, int healthPoint, int damage) : Unit(name, healthPoint, damage)  {
    setAbility(new RogueAbility(this));
    setCurrentState(new HumanState(this));
    setNextState(new HumanState(this));
    setUnitType(rogue);
}

Rogue::~Rogue() {}

File: Army-Java/Units/Rogue.java
package com.gymfox.Army.Units;

import com.gymfox.Army.Ability.RogueAbility;
public class Rogue extends Unit {
    public Rogue(String name, int healthPointLimit, int damage) {
        super(name, healthPointLimit, damage);
        this.ability = new RogueAbility(this);
    }
}
